pkg/metrics/handler.go

```diff
@@ -0,0 +1,51 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/rancher/rancher/pkg/auth/util"
+	v2 "k8s.io/api/authorization/v1"
+	v3 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+type metricsHandler struct {
+	subjectAccessReviewClient v1.SubjectAccessReviewInterface
+	promHandler               http.Handler
+}
+
+// NewMetricsHandler ensures the user is allowed to read the Prometheus metrics endpoint
+func NewMetricsHandler(scaledContextClient kubernetes.Interface, promHandler http.Handler) http.Handler {
+	return &metricsHandler{
+		subjectAccessReviewClient: scaledContextClient.AuthorizationV1().SubjectAccessReviews(),
+		promHandler:               promHandler,
```

Comment:
If this was me, I'd pass in a `v1.SubjectAccessReviewInterface` rather than the `kubernetes.Interface` ?